migrations/2017-10-08-193512_category_trees/up.sql

```diff
@@ -0,0 +1,40 @@
+-- Your SQL goes here
+CREATE EXTENSION ltree;
+
+-- Create the new column which will represent our category tree.
+-- Fill it with values from `slug` column and then set to non-null
+ALTER TABLE categories
+    ADD COLUMN path ltree;
+
+-- Unfortunately, hyphens (dashes) are not allowed...
+UPDATE categories
+    SET path =  text2ltree('root.' || trim(replace(replace(slug, '-', '_'), '::', '.')))
+    WHERE path is NULL;
+
+ALTER TABLE CATEGORIES
+    ALTER COLUMN path SET NOT NULL;
+
+-- Create some indices that allow us to use GIST operators: '@>', etc
+CREATE INDEX path_gist_categories_idx ON categories USING GIST(path);
+CREATE INDEX path_categories_idx ON categories USING btree(path);
```

Comment:
Is there a reason to have both a gist and a btree index? The [postgres docs](https://www.postgresql.org/docs/9.6/static/ltree.html) say:

> F.21.3. Indexes
> ltree supports several types of indexes that can speed up the indicated operators:
>
> * B-tree index over ltree: <, <=, =, >=, >
> * GiST index over ltree: <, <=, =, >=, >, @>, <@, @, ~, ?

which looks like gist indexes take care of the same operators that btree indexes take care of? 

All of this is probably moot anyway since the categories table is so smalll :)

Reply:
Honestly, I've just seen everyone do it that way, so I figured it was a best practice, if not necessary.